File: FizzBuzz/FizzBuzz.py
# coding: utf-8

import os
from distutils.util import strtobool

def FizzBuzz():
    while True:
        try:
            option = strtobool(input("\nHello! Would you like to play FizzBuzz? (y/n) "))
        except ValueError:
            print("\nSorry, I didn't understand!\nYou should reply with yes or no!\n")
            os.system("pause")
            os.system('cls')
        else:
            if option == 1:
                os.system('cls')
                print("\nGreat! I love this game!\n")
                min, max = get_range()
                print("\nAwsome! Now that we have our start and end we need our fizz and our buzz!\n")
                fizz, buzz = get_fizzbuzz()
                for i in range(min, max+1):
                    if i % fizz == 0 and i % buzz == 0:
                        print("FizzBuzz")
                    elif i % fizz == 0:
                        print("Fizz")
                    elif i % buzz == 0:
                        print("Buzz")
                    else:
                        print(i)
                try:
                    option = strtobool(input("\nWould you like to play again? (y/n) "))
                except ValueError:
                    print("\nSorry, I didn't understand!\nYou should reply with yes or no!\n")
                    os.system("pause")
                    os.system('cls')
            elif option == 0:
                os.system('cls')
                print("\nThat's too bad, let's play another time then!\nGoodbye!\n")
                os.system("pause")
                quit()          

def get_range():
    while True:
        print("\nTo start, we will need to set the range!\n\nRemember both the begining and the end should be Integers,\nand the end should be higher than the start!\n")
        os.system("pause")
        os.system('cls')
        print("What is the start number?")
        min = get_int()
        print("Great! We will start at", min,"then!\n")
        print("What is the end number?")
        max = get_int()
        while max <= min:
            print("That doesn't seem right... the game can't end before it starts!\nMake sure you're writing a number that is larger than", min,"!\n\nLet's try that again! When should the game end? ")
            max = get_int()
        print("Great! We will end with", max,", excellent choice!\n")
        os.system("pause")
        os.system('cls')
        return min, max

def get_fizzbuzz():
    while True:
        print("What should fizz be?")
        fizz = get_int()
        while fizz == 0:
            os.system('cls')
            print("Well, the only multiple of 0 is 0 so thats no fun... Can you pick a diferent number?\n")
            print("What should fizz be?")
            fizz = get_int()
        print("Awsome! We're gonna replace multiples of",fizz,"with the word Fizz!\n")
        os.system("pause")
        os.system('cls')
        print("And what should buzz be?")
        buzz = get_int()
        while buzz == 0 or buzz == fizz:
            os.system('cls')
            if buzz == 0:
                print("Well, the only multiple of 0 is 0 so thats no fun... Can you pick a diferent number?\n")
            if buzz == fizz:
                print("For the most fun buzz and fizz should be different numbers! Can you pick a diferent number?\n")
            print("What should buzz be?")
            buzz = get_int()
        print("Awsome! We're gonna replace multiples of",buzz,"with the word Buzz!\n")
        print("Multiples of both",fizz,"and",buzz,"will be replaced with FizzBuzz!\n")
        os.system("pause")
        os.system('cls')
        return fizz, buzz

def get_int():
    while True:
        try:
            num = int(input())
            print()
            return num
        except ValueError:
            print("\nSorry, I didn't understand!\nYou should reply with an Integer (whole number)!\n")
            os.system("pause")
            os.system('cls')

os.system('cls')
FizzBuzz()

File: FizzBuzz/FizzBuzz.cpp
#include <iostream>

using namespace std;

struct AB
{
    int a;
    int b;
};

bool getYN(string answer);
void fizzbuzz();
AB getAB();

int main()
{
    string answer;
    while (true)
    {
        cout << "Hello! Would you like to play FizzBuzz? (y/n) ";
        cin >> answer;
        if (getYN(answer) == true)
        {
            fizzbuzz();
        }
        else
        {
            cout << "That's too bad, let's play another time then!" << endl << "Goodbye!";
            return 0;
        }
    }
}

//validate yes or no replies
bool getYN(string answer)
{
    while (true)
    {
        answer[0] = toupper(answer[0]);
        if (answer[0] == 'Y' || answer[0] == '1')
        {
            return true;
        }
        else if (answer[0] == 'N' || answer[0] == '0')
        {
            return false;
        }
        else
        {
            cout << "Sorry, I didn't understand!" << endl << "You should reply with yes or no!" << endl << "Would you like to play FizzBuzz? (y/n) ";
            cin >> answer;
        }
    }
}

void fizzbuzz()
{
    cout << "Great! I love this game!" << endl << "First we need to set the range!" << endl << "We need two numbers, one for the start and one for the end." << endl << "Make sure you only insert integers (whole numbers) over 0!" << endl;    
    AB range, fizzbuzz;
    do
    {
        range = getAB();
        if (range.a > range.b)
        {
            cout << "That doesn't seem right... the game can't end before it starts!" << endl << "Make sure you're writing a second number that is larger than the first" << endl;
        }
    } while (range.a > range.b);
    cout << "Awsome!" << endl << "We will start with " << range.a << " and end with " << range.b << "!" << endl;
    do
    {
        fizzbuzz = getAB();
        if (fizzbuzz.a == fizzbuzz.b)
        {
            cout << "That doesn't seem right... we cant use the same number twice!" << endl << "Make sure you're writing a second number that is diferent than the first" << endl;
        }
    } while (fizzbuzz.a == fizzbuzz.b);
    cout << "Great! We're gonna replace multiples of " << fizzbuzz.a << "with the word fizz and multiples of " << fizzbuzz.b << " with the word buzz!" << endl << "Multiples of both will be replaced with FizzBuzz!" << endl;
    for (int i = range.a; i <= range.b; i++)
    {
        if (i % fizzbuzz.a == 0 || i % fizzbuzz.b == 0)
        {
            if (i % fizzbuzz.a == 0 && i % fizzbuzz.b == 0)
            {
                cout << "FizzBuzz!" << endl;
            }
            else if (i % fizzbuzz.a == 0 && i % fizzbuzz.b != 0)
            {
                cout << "Fizz!" << endl;
            }
            else if (i % fizzbuzz.a != 0 && i % fizzbuzz.b == 0)
            {
                cout << "Buzz!" << endl;
            }
        }
        else
        {
            cout << i << endl;
        }
    }
}

AB getAB()
{
    AB pair;
    do
    {
        cout << "What should the first number be? ";
        cin >> pair.a;
        if (pair.a < 1)
        {
            cout << "Sorry, it has to be an integer over 0!" << endl;
        }
    } while (pair.a < 1);
    do
    {
        cout << "What should the second number be? ";
        cin >> pair.b;
        if (pair.b < 1)
        {
            cout << "Sorry, it has to be an integer over 0!" << endl;
        }
    } while (pair.b < 1);
    return pair;
}